src/control/group/sse2.rs

```diff
@@ -124,22 +134,16 @@ impl Group {
     /// - `FULL => DELETED`
     #[inline]
     pub(crate) fn convert_special_to_empty_and_full_to_deleted(self) -> Self {
-        // Map high_bit = 1 (EMPTY or DELETED) to 1111_1111
-        // and high_bit = 0 (FULL) to 1000_0000
-        //
-        // Here's this logic expanded to concrete values:
-        //   let special = 0 > tag = 1111_1111 (true) or 0000_0000 (false)
-        //   1111_1111 | 1000_0000 = 1111_1111
-        //   0000_0000 | 1000_0000 = 1000_0000
-        #[allow(
-            clippy::cast_possible_wrap, // tag: Tag::DELETED.0 as i8
-        )]
+        debug_assert_eq!(127, Tag::EMPTY.0);
+        debug_assert_eq!(126, Tag::DELETED.0);
+
+        #[allow(clippy::cast_sign_loss, clippy::cast_possible_truncation)]
         unsafe {
-            let zero = x86::_mm_setzero_si128();
-            let special = x86::_mm_cmpgt_epi8(zero, self.0);
+            let is_special = x86::_mm_cmpgt_epi8(self.0, x86::_mm_set1_epi32(Tag::MAX_TAG32));
+
             Group(x86::_mm_or_si128(
-                special,
-                x86::_mm_set1_epi8(Tag::DELETED.0 as i8),
+                x86::_mm_and_si128(is_special, x86::_mm_set1_epi32(Tag::EMPTY32)), // EMPTY if special
+                x86::_mm_set1_epi32(Tag::DELETED32), // else DELETED
```

Comment:
You can simplify this: the compare gives you `0` and `-1`, to which you can just add `127` to get `EMPTY` and `DELETED`.